extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -466,26 +486,39 @@ private function processAllInTrack() {
 				}
 			}
 		}
+	}
 
-
-		// BADGE_LOVE
-		if(empty($this->mCounters[BADGE_LOVE])) {
+	private function processInTrackLove() {
+		//BADGE_LOVE
+		if (empty($this->mCounters[BADGE_LOVE])) {
 			$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER] = 1;
 		} else {
-			if($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d')) {
+			if ($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d')) {
 				// ignore
-			} else if($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d', strtotime('-1 day'))) {
+			} else if ($this->mCounters[BADGE_LOVE][COUNTERS_DATE] == date('Y-m-d', strtotime('-1 day'))) {
 				$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER]++;
 			} else {
 				$this->mCounters[BADGE_LOVE][COUNTERS_COUNTER] = 1;
 			}
 		}
 		$this->mCounters[BADGE_LOVE][COUNTERS_DATE] = date('Y-m-d');
+	}
 
-		wfProfileOut(__METHOD__);
+	private function incrementCounterBy($badge_type_id, $value) {
+		$this->mCounters = is_array($this->mCounters) ? $this->mCounters : $this->mUserCountersService->getCounters();
+		if (empty($this->mCounters[$badge_type_id])) {
```

Comment:
So the code is refactored and the old code was using empty(). But after you pointed out, I investigated and !isset() makes more sense because 0 is true to empty(). The code was if(empty) then set to 0. Then, 0 will be evaluated to empty() again, kind of waste.